analysis_tools/src/HistogramFiller.cpp

```diff
@@ -38,12 +39,17 @@ void HistogramFiller::fill_hists(const Particle* particle)
   if(particle->is_kaon()) histSets["single_kaons"]->fill(particle);
 }
 
-void HistogramFiller::fill_hists(const Particle* particle, const Particle* particle2)
+void HistogramFiller::fill_hists(const Particle* particle1, const Particle* particle2)
 {
-  if(!particle || !particle2) return;
+  if(!particle1 || !particle2) return;
   // pi+ and pi-
-  if(particle->is_pion() && particle2->is_pion() && particle->pdgid == -1*particle2->pdgid && particle->is_final() && particle2->is_final()){
-    histSets["double_pions"]->fill(particle, particle2);
+  if(particle1->is_pion() && particle2->is_pion() && particle1->pdgid == -1*particle2->pdgid && particle1->is_final() && particle2->is_final()){
+    histSets["double_pions"]->fill(particle1, particle2);
+  }
+
+  if(particle1->is_pion() && particle2->is_pion() && particle1->pdgid == -1*particle2->pdgid && particle1->is_final() && particle2->is_final() &&
+     particle1->has_mother_with_id(113) && particle2->has_mother_with_id(113)){
+    histSets["double_pions_from_rho_decay"]->fill(particle1, particle2);
```

Comment:
I would propose to merge these two ifs:

```C++
if(particle1->is_pion() && particle2->is_pion() && 
   particle1->pdgid == -1*particle2->pdgid && 
   particle1->is_final() && particle2->is_final())
{
     histSets["double_pions"]->fill(particle1, particle2);
     
     if(particle1->has_mother_with_id(113) && particle2->has_mother_with_id(113)){
          histSets["double_pions_from_rho_decay"]->fill(particle1, particle2);
     }
}

```

Reply:
Ok, I pushed new commit with this change.